### ISSUE TITLE
Not able to reproduce with DGL Problems

### ISSUE THREAD

**Issue by VityaVitalich** [minhaoJ2/TaxoEnrich#6]
Dear maintainers, i would like to reproduce your paper results and dive deeper into the taxonomy enrichment. However i face an issue with running training scripts. It seems like some dependencies are broken. 

For example if i am using dgl > 0.9.0 i am facing the following error
AttributeError: Can't get attribute 'DGLHeteroGraph' on <module 'dgl.heterograph' from '/root/anaconda3/envs/myenv/lib/python3.8/site-packages/dgl/heterograph.py'>

this happens on the reading of pickle bin file at line 152 in dataset.py

Otherwise if i am trying lower version of DGL, i am facing the following error in exact same place
UnpicklingError: pickle data was truncated

Would you please help me to fix those issues? maybe publishing requirements file with proper python version would be the easiest solution

**Comment by minhaoJ2** [comment]
Hi,

I used Python 3.6 and the default DGL version for the experiment environment at the time of the paper. This could be related to the new features of the DGL version, which I'm unfamiliar with.

**Comment by VityaVitalich** [comment]
I have tried iterating over all versions of DGL library https://pypi.org/project/dgl/#history , and have faced different issues connected with it on every version using python 3.6

Is it possible to find the appropriate version of DGL to work with the code in repo?

**Comment by minhaoJ2** [comment]
The original setup with Python 3.6 worked on my end. Different cuda version may require different library versions. We also observed some instabilities and discrepancies for different machines.

**Comment by VityaVitalich** [comment]
Sure, i am trying to reproduce the code, so could you please provide the cuda versions and suitable libraries in any type of requirements.txt
Otherwise, the reproduction looks like brute-forcing over all possible libraries versions

**Comment by minhaoJ2** [comment]
iirc, the cuda version I used for the experiments was cuda 10.0 or higher (not sure which version exactly since the experiments were finished a long time ago.) Other libraries are the same version as stated in requirements.txt. Sorry cannot help more.

**Comment by VityaVitalich** [comment]
Could you please help with requirements.txt
seems like it is missing in repo

**Comment by minhaoJ2** [comment]
Sorry for the confusion. That's right I do not have requirements.txt for this project. Then all the packages I used were the newest version for the names I listed in README. Other packages that I remember the specific version numbers are provided above.

**Comment by VityaVitalich** [comment]
I afraid that there is some confusion.

For example, running embedding_generation produces the following error with gensim. 
```Traceback (most recent call last):
File "embedding_generation.py", line 4, in  from gensim.models.wrappers import FastText
ModuleNotFoundError: No module named 'gensim.models.wrappers' ```

However i have installed 4.1.2 version of gensim, that was released on 17.09.2021. That was the newest version to the moment that code in repo was published (8.02.2022)

**Comment by minhaoJ2** [comment]
Probably gensim wrappers were replaced with some other modules in the new version. You can refer to the documentation or other forums to see how to fix this.
